### PR TITLE
Leverage Generated Reconciler for reconciliation.

### PR DESCRIPTION
**Pull request by n3wscott**
Using the new `// +genreconciler` in sample-source. Tested working with events produced.

### CODE REVIEW

File: config/500-controller.yaml [hunk 1]
```diff
@@ -13,48 +13,52 @@
...
+    matchLabels:
+      app: sample-source-controller
   template:
     metadata:
-      labels: *labels
+      labels:
+        app: sample-source-controller
+        samples.knative.dev/release: devel
     spec:
       serviceAccountName: sample-source-controller-manager
```
Comment (matzew) [inline config/500-controller.yaml]: should this be renamed ? Including the SA ?

File: pkg/reconciler/eventtypes.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	"knative.dev/pkg/logging"
+
+	"knative.dev/sample-source/pkg/apis/samples/v1alpha1"
```
Comment (antoineco) [inline pkg/reconciler/eventtypes.go]: Same comment about aliasing this to `samplesv1alpha1`

File: pkg/reconciler/controller.go [hunk 3]
```diff
@@ -18,27 +18,23 @@ package reconciler
...
-)
-
-const (
-	// ReconcilerName is the name of the reconciler
-	ReconcilerName = "SampleSource"
-
-	// controllerAgentName is the string used by this controller to identify
-	// itself when creating events.
-	controllerAgentName = "sample-source-controller"
+	"knative.dev/sample-source/pkg/client/injection/reconciler/samples/v1alpha1/samplesource"
```
Comment (antoineco) [inline pkg/reconciler/controller.go]: Same comment about aliasing to `samplesourceclient`

File: pkg/reconciler/controller.go [hunk 4]
```diff
@@ -18,27 +18,23 @@ package reconciler
 
 import (
 	"context"
+	"knative.dev/pkg/logging"
```
Comment (antoineco) [inline pkg/reconciler/controller.go]: This import somehow got separated from other knative imports

File: config/500-controller.yaml (hunk 1 above)
Comment (n3wscott) [reply config/500-controller.yaml]: likely, but not in my PR

File: pkg/reconciler/samplesource.go [hunk 5]
```diff
@@ -31,46 +28,62 @@ import (
...
-	"knative.dev/eventing/pkg/logging"
-	"knative.dev/eventing/pkg/reconciler"
 	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
 	"knative.dev/pkg/resolver"
 
 	"knative.dev/sample-source/pkg/apis/samples/v1alpha1"
 	versioned "knative.dev/sample-source/pkg/client/clientset/versioned"
+	"knative.dev/sample-source/pkg/client/injection/reconciler/samples/v1alpha1/samplesource"
```
[the remark this answers is not in the document]
Comment (n3wscott) [reply pkg/reconciler/samplesource.go]: maybe `reconcilersamplesource`

File: pkg/reconciler/samplesource.go [hunk 6]
```diff
@@ -31,46 +28,62 @@ import (
...
+	eventingclientset "knative.dev/eventing/pkg/client/clientset/versioned"
 	eventinglisters "knative.dev/eventing/pkg/client/listers/eventing/v1alpha1"
-	"knative.dev/eventing/pkg/logging"
-	"knative.dev/eventing/pkg/reconciler"
 	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
 	"knative.dev/pkg/resolver"
 
 	"knative.dev/sample-source/pkg/apis/samples/v1alpha1"
```
[the remark this answers is not in the document]
Comment (n3wscott) [reply pkg/reconciler/samplesource.go]: we do not alias the imports when it is for the "first class package" of the repo

File: pkg/reconciler/eventtypes.go (hunk 2 above)
Comment (n3wscott) [reply pkg/reconciler/eventtypes.go]: we don't do this normally for repos like this.

File: pkg/reconciler/controller.go (hunk 4 above)
Comment (n3wscott) [reply pkg/reconciler/controller.go]: goland

Comment (antoineco) [reply pkg/reconciler/controller.go]: I'm not convinced this is a reason not to make it right.

File: pkg/reconciler/eventtypes.go (hunk 2 above)
Comment (antoineco) [reply pkg/reconciler/eventtypes.go]: Which is alright for this minimal example, but aliasing APIs is a common Kubernetes practice no matter the context. Besides it helps identifying the groups of objects in the code and is a necessary evil when you start having multiple API groups

File: pkg/reconciler/samplesource.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (antoineco) [reply pkg/reconciler/samplesource.go]: That's what I meant, sorry!

**Pull request merged**